### PR TITLE
Add pre-commit hook and basic test

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a pre-commit hook and basic test.

### CODE REVIEW

**Comment by oyaron** [comment]
@mcoughlin Hi Michael. Does this include only cosmetics/cleaning or is there some change of logic?

**Comment by mcoughlin** [comment]
This is basically running pre-commit and adding that test. No logic changes outside of need to fix import styles to meet flake8

File: NGSF/SF_functions.py [hunk 1]
```diff
@@ -444,128 +437,117 @@ def all_parameter_space(int_obj,redshift, extconstant, templates_sn_trunc, templ
...
-    print('Runtime: {0: .2f}s '.format(end-start))
+    # if plot==1:
 
-
-
-    #if plot==1:
-        
     #    for i in range(0,n):
 
     #        plotting(int_obj,result[:][i], lam , original, i, save=save, show=show)
```
Comment (guynir42) [inline NGSF/SF_functions.py:551]:
```suggestion
```

File: NGSF/SF_functions.py [hunk 2]
```diff
@@ -444,128 +437,117 @@ def all_parameter_space(int_obj,redshift, extconstant, templates_sn_trunc, templ
...
 
-    return 
-
+    return
```
Comment (guynir42) [inline NGSF/SF_functions.py:553]: Do we need this return at the end?

File: NGSF/error_routines.py [hunk 3]
```diff
@@ -1,94 +1,87 @@
...
+        return ret[n - 1 :] / n
+
     mov_var = moving_average(resid**2, n=100)
-    mov_var = np.concatenate((mov_var, [mov_var[-1]]* (resid.size-mov_var.size)))
-    err_std = mov_var**(1/2)
-    #print(err_std)
-    
+    mov_var = np.concatenate((mov_var, [mov_var[-1]] * (resid.size - mov_var.size)))
+    err_std = mov_var ** (1 / 2)
+    # print(err_std)
```
Comment (guynir42) [inline NGSF/error_routines.py:74]:
```suggestion
```

File: NGSF/error_routines.py [hunk 4]
```diff
@@ -1,94 +1,87 @@
...
-    
+        return ret[n - 1 :] / n
+
     mov_var = moving_average(resid**2, n=100)
-    mov_var = np.concatenate((mov_var, [mov_var[-1]]* (resid.size-mov_var.size)))
-    err_std = mov_var**(1/2)
-    #print(err_std)
-    
+    mov_var = np.concatenate((mov_var, [mov_var[-1]] * (resid.size - mov_var.size)))
+    err_std = mov_var ** (1 / 2)
```
Comment (guynir42) [inline NGSF/error_routines.py:73]:
```suggestion
    err_std = np.sqrt(mov_var)
```

File: NGSF/error_routines.py [hunk 5]
```diff
@@ -1,94 +1,87 @@
...
-    #x = [i+1e-10 for i in x if i==0]
-    #err_std = [i+1e-10 for i in err_std if i==0]
-    #err_std= [Decimal(i) for i in err_std if i==0]
-    #for i in range(0,len(x)):
+    # x = [i+1e-10 for i in x if i==0]
+    # err_std = [i+1e-10 for i in err_std if i==0]
+    # err_std= [Decimal(i) for i in err_std if i==0]
+    # for i in range(0,len(x)):
     #    if i==0:
     #        x[i] = 'nan'
```
Comment (guynir42) [inline NGSF/error_routines.py:85]:
```suggestion
```

File: NGSF/get_metadata.py [hunk 6]
```diff
@@ -4,130 +4,133 @@
...
+class Metadata(object):
 
+    def __init__(self):
 
+        parameters = Parameters(data)
 
-with open(mjd_max_brightness, mode='r') as inp:
-    reader = csv.reader(inp)
-    band_dictionary = {rows[0]:rows[2] for rows in reader}
+        mydict = {}
```
Comment (guynir42) [inline NGSF/get_metadata.py:31]: Do we use `mydict` anywhere?

File: NGSF/params.py [hunk 7]
```diff
@@ -3,167 +3,168 @@
...
+        self.object_to_fit = data["object_to_fit"]
+        self.save_results_path = data["saving_results_path"]
+
+        self.use_exact_z = data["use_exact_z"]
+        self.z_exact = data["z_exact"]
+        self.z_range_begin = data["z_range_begin"]
+        self.z_range_end = data["z_range_end"]
+        self.z_int = data["z_int"]
+
+        if self.use_exact_z == 1:
```
Comment (guynir42) [inline NGSF/params.py:58]:
```suggestion
        if self.use_exact_z:
```

**Review by guynir42 — APPROVED** [review]
I just wanted to verify that most of the files have only gotten cosmetic changes. It looks fine to me. Of course the repo admins should have a look and approve this rather big PR.

File: .github/workflows/python-package.yml [hunk 8]
```diff
@@ -0,0 +1,52 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Unit Test NGSF
+
+on: [push]
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply .github/workflows/python-package.yml:6]: I guess this is updated once we update the branch?

File: NGSF/get_metadata.py (hunk 6 above)
Comment (mcoughlin) [reply NGSF/get_metadata.py]: I removed it

File: .github/workflows/python-package.yml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics --exclude docs
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics --exclude docs,versioneer.py,NGSF/_version.py,NGSF/tests,NGSF/*/__init__.py
+    - name: Test with pytest
+      run: |
+        pwd
+        ls
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply .github/workflows/python-package.yml]: I removed it.

**Comment by mcoughlin** [comment]
@oyaron Did you have a chance to do a quick test to be sure this doesn't break anything for you? At least it now has a test.

**Comment by oyaron** [comment]
@mcoughlin Hopefully we'll clarify and merge everything in the coming day or two.

**Pull request merged**